m4/autoconf-archive/ax_check_compile_flag.m4

```diff
@@ -0,0 +1,72 @@
+# ===========================================================================
+#   http://www.gnu.org/software/autoconf-archive/ax_check_compile_flag.html
+# ===========================================================================
```

Comment:
@lumag yes, maybe message is not really clear. autoconf is not depends on cross tools and depends on host env only. Idea was that for some env this package exit. For some other it is not. I see that Linaro CI fails. Or I need to update all docker images if it's possible or just put this file here.

If we already have autoconf-archive in m4/ folder than it's really good to be there. I my patch I wanted to put it to separate folder to be more clear that it's related to autoconf-archive package.